cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -1318,6 +1324,29 @@ updateProposalSupportedInEra AlonzoEra  = Just UpdateProposalInAlonzoEra
 updateProposalSupportedInEra BabbageEra = Just UpdateProposalInBabbageEra
 updateProposalSupportedInEra ConwayEra  = Just UpdateProposalInConwayEra
 
+-- | A representation of whether the era supports the 'UTxO Cost Per Byte'
+-- protocol parameter.
+--
+-- The Babbage and subsequent eras support such a protocol parameter.
+--
+data ProtocolUTxOCostPerByteSupportedInEra era where
```

Comment:
So we actually explicitly avoided using GADTs for `ProtocolParameter` fields because it would result in an explosion of type definitions.

Reply:
I added GADTs because that was the way we handle era support for things that aren't protocol parameters.

The cost will be `(2 x numEras + 5) x numRelevantProtocolPameters`.  I don't yet have a good idea how many protocol parameters need this.  At least each new protocol parameter will need this.

The ability to determine what protocol parameters are available for what era will be important to generators but I also suspect users of our API will find this useful too.

As for what alternative solutions are available, it will need to be one that allows us to express both the addition and removal of protocol parameters because new eras can and have removed protocol parameters.

If not expressed as types it could be expressed as a runtime value.

Reply:
For comparison:

* Number of features: `24`
* Number of protocol parameters: `25`

Reply:
I've updated PR with an more concise encoding.  This means we write less code when adding a new parameter or feature.

We could also convert the existing feature supported in era code to use this new system instead which would reduce the amount of code.